### PR TITLE
Terms and Conditions for Medical Information

### PR DESCRIPTION
**Pull request by U-DON**
Resolves department-of-veterans-affairs/vets.gov-team#10051.

### CODE REVIEW

File: src/platform/user/authorization/containers/MHVApp.jsx
Comment (U-DON) [inline src/platform/user/authorization/containers/MHVApp.jsx]: This goes to the T&C page with the current health tool's URL as a query param. That allows acceptance to redirect back here and use another query param as a flag to trigger a success message.

File: src/platform/user/authorization/containers/MHVApp.jsx
Comment (U-DON) [inline src/platform/user/authorization/containers/MHVApp.jsx]: This goes to the T&C page with the current health tool's URL as a query param. That allows acceptance to redirect back here and use another query param as a flag to trigger a success message.

**Review by bshyong — APPROVED** [review]

**Review by dhall — APPROVED** [review]
LGTM!

File: src/js/terms-and-conditions/containers/MhvTermsAndConditions.jsx
Comment (ncksllvn) [inline src/js/terms-and-conditions/containers/MhvTermsAndConditions.jsx]: Would it be worth returning a promise from the `acceptTerms` action and using `setState` in a `.then`?  I ran it locally and the API endpoint wasn't configured so it returned a `404`, but it was kinda confusing how it still showed the success message.

File: src/js/terms-and-conditions/containers/MhvTermsAndConditions.jsx
Comment (ncksllvn) [inline src/js/terms-and-conditions/containers/MhvTermsAndConditions.jsx]: Should we add anything here to account for errors stored in `this.props.errors`? I see that added into the reducer but don't think it's being rendered.

File: src/js/terms-and-conditions/actions/index.js
Comment (ncksllvn) [inline src/js/terms-and-conditions/actions/index.js]: `apiRequest` does return a Promise in case you want to return it and use it as a quick method of showing success/failure state that aren't worth storing in the store.

**Review by ncksllvn — APPROVED** [review]
LGTM

File: src/js/terms-and-conditions/containers/MhvTermsAndConditions.jsx
Comment (U-DON) [reply src/js/terms-and-conditions/containers/MhvTermsAndConditions.jsx]: Good point. We probably don't want to show the success before the API returns with a success. Usually I would handle this in `componentDidUpdate`.

```
  componentDidUpdate() {
    if (!prevProps.accepted && this.props.accepted) {
      this.setState({ showAcceptedMessage: true }, () => {
        scroller.scrollTo('banner', getScrollOptions());
        this.redirect();
      });
    }
  }
```

File: src/js/terms-and-conditions/containers/MhvTermsAndConditions.jsx
Comment (U-DON) [reply src/js/terms-and-conditions/containers/MhvTermsAndConditions.jsx]: We probably should. Don't have design/copy for it yet, though.

File: src/js/terms-and-conditions/containers/MhvTermsAndConditions.jsx
Comment (ncksllvn) [reply src/js/terms-and-conditions/containers/MhvTermsAndConditions.jsx]: Ah nice! That's a better way to do it than my idea but I still miss the days where you just used Promises for situations like that.

**Comment by BBarone1** [comment]
subscribing to this thread as this change will impact google analytics tracking for personalization once merged in

**Comment by U-DON** [comment]
@melwoodard: A warning message would seem to make more sense as a response to Cancel. Redirecting to go back could work too, but I will defer to @ehuntdsva on the design decision.

**Comment by ehuntdsva** [comment]
@U-DON Apologies - I was working too quickly on the designs and forgot to change the BG color on that banner. @melwoodard is correct - on the "after cancelling T&C" screen, it should be the yellow "warning" BG color. 

@melwoodard for reference, here are the user flows.
flows_v2

**Pull request merged**
